### DISCUSSION TITLE
[Feature request] Add a "Randomize Watchlist" feature

### DISCUSSION THREAD

**Discussion by anks-mrangry** [trakt/showly#902] [Ideas / Feature Requests]
@michaldrabik
**Description**:
When I have a long watchlist and I'm not sure what to watch, it would be helpful to have a "Randomize Watchlist" feature that would randomly sort my watchlist. This would help me discover new things to watch and avoid decision fatigue.

**Additional context:**
- This feature would be especially useful for users with large watchlists.
- The randomization algorithm could be seeded with the current date and time to ensure that the watchlist is randomized differently 
each time.
- This feature could be added to the "Watchlist" section of the app.

**Acceptance criteria:**
- The "Randomize Watchlist" feature is available in the app.
- The watchlist is randomized when the user taps the "Randomize" button.
- The randomization algorithm is seeded with the current date and time.
- The randomized watchlist is displayed in the "Watchlist" section of the app.

**Comment by michaldrabik** [comment]
Would you say it's enough to add another sorting type of "Random"? The list would then be sorted randomly whenever it is opened.

**Reply by anks-mrangry** [reply]
@michaldrabik 
Yes, you're absolutely right. Adding a "Random" sorting option that persists like other sorting options would be a great solution. This way, users can choose to have their watchlist randomly sorted until they decide to change it.

**_Here's a breakdown of how this would work:_**

**Add "Random" to the Sorting Options:**
In the watchlist screen, add "Random" as a new sorting option (for both Shows & Movies) alongside "Released," "Genres," etc.

**Persistent Sorting:**
When a user selects "Random," the watchlist is shuffled and displayed in a random order.
This random order is maintained until the user changes the sorting option or closes and reopens the watchlist.

**Clear Sorting:**
To revert to the default sorting or another specific sorting option, the user can simply unselect Random option.

This approach provides a flexible and user-friendly way to _discover new content within the watchlist._

![393521157-eb7b6a86-dac8-42c5-abc0-d9830a537a20](https://github.com/user-attachments/assets/ee94f392-d2b3-4eef-b7a2-8a988f69eee2)
![393521734-425b25f5-ad58-4309-84c4-6665e8faeec1](https://github.com/user-attachments/assets/5b82ba53-d360-4e47-b22e-ac50268ad32e)

**Reply by anks-mrangry** [reply]
Hi @michaldrabik,

I hope this message finds you well.

I'm following up on my previous comment regarding the "Randomize Watchlist" feature request. I'm eager to know if this feature is being considered for future updates.

Any information on the potential timeline or priority for this feature would be greatly appreciated.

Thank you for your time and consideration.

**Reply by michaldrabik** [reply]
@anks-mrangry I have actually added this to my personal roadmap as it seems nice addition. However as you may notice I'm never promising any particular release dates or timelines because, since I am alone, its hard to always keep all promises. Stay tuned!

**Reply by anks-mrangry** [reply]
@michaldrabik 
Thank you so much for taking the time to consider my request and for adding it to your roadmap! I understand that you're working solo, and I appreciate your efforts.

I'm excited to see what you come up with and will definitely stay tuned for updates.

Thanks again!
